pyk/src/pyk/cterm/show.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from ..kast.inner import KApply, KSort, KToken, flatten_label, top_down
+from ..kast.manip import free_vars, minimize_term
+from ..kast.prelude.k import DOTS
+from .cterm import CTerm
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable
+    from typing import Final
+
+    from ..kast.inner import KInner
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+class CTermShow:
+    _printer: Callable[[KInner], list[str]]
```

Comment:
Consider

```suggestion
    _printer: Callable[[KInner], str]
```

Reason:
* It's a simpler interface.
* Both `Formatter` and `PrettyPrniter` produce `str`. (In fact, `Formatter` is already a `Callable[[KInner], str]`)

Then `print` can be replaced by `_print_lines` which does the split.

Reply:
Done.